### ISSUE TITLE
WP/CronInterval: incorrect handling of namespaced first-class callables

### ISSUE THREAD

**Issue by rodrigoprimo** [WordPress/WordPress-Coding-Standards#2644]
## Bug Description

The `WordPress.WP.CronInterval` sniff incorrectly handles namespaced first-class callables (PHP 8.1+) passed as callbacks to `add_filter()`. It treats namespaced function references as if they were global functions with the same name, leading to both false positives and false negatives.

### Problem 1: false negatives

```php
// Global function with a long interval
function first_class_weekly_schedule( $schedules ) {
    $schedules['weekly'] = array(
        'interval' => WEEK_IN_SECONDS,
        'display' => __( 'Once Weekly' )
    );
    return $schedules;
}

// These should trigger ChangeDetected warning but don't trigger any warning
add_filter( 'cron_schedules', MyNamespace\first_class_weekly_schedule(...));
add_filter( 'cron_schedules', \MyNamespace\first_class_weekly_schedule(...));
```

When running the sniff against this code snippet, I expect to see two `WordPress.WP.CronInterval.ChangeDetected` warnings, but instead there are no warnings.

### Problem 2: false positives

```php
// Global function with a short interval
function first_class_six_min_schedule( $schedules ) {
    $schedules['every_6_mins'] = array(
        'interval' => 360,
        'display' => __( 'Once every 6 minutes' )
    );
    return $schedules;
}

// These should trigger ChangeDetected but trigger CronSchedulesInterval instead
add_filter( 'cron_schedules', MyNamespace\first_class_six_min_schedule(...));
add_filter( 'cron_schedules', \MyNamespace\first_class_six_min_schedule(...));
add_filter( 'cron_schedules', namespace\Sub\first_class_six_min_schedule(...));
```

When running the sniff against this code snippet, I expect to see three `WordPress.WP.CronInterval.ChangeDetected` warnings, but instead I get three `WordPress.WP.CronInterval.CronSchedulesInterval` warnings.

```bash
$ vendor/bin/phpcs -s test.php --standard=WordPress --sniffs=WordPress.WP.CronInterval test.php                                                                                                                                           at 15:19:02 

FILE: test.php
---------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 3 WARNINGS AFFECTING 3 LINES
---------------------------------------------------------------------------------------------------------------------------------------
 13 | WARNING | Scheduling crons at 360 sec ( less than 15 minutes ) is discouraged. (WordPress.WP.CronInterval.CronSchedulesInterval)
 14 | WARNING | Scheduling crons at 360 sec ( less than 15 minutes ) is discouraged. (WordPress.WP.CronInterval.CronSchedulesInterval)
 15 | WARNING | Scheduling crons at 360 sec ( less than 15 minutes ) is discouraged. (WordPress.WP.CronInterval.CronSchedulesInterval)
---------------------------------------------------------------------------------------------------------------------------------------
```

## Environment
| Question                 | Answer
| ------------------------ | -------
| PHP version              | 8.4
| PHP_CodeSniffer version  | 3.13.4
| WordPressCS version      | develop
| PHPCSUtils version       | 
| PHPCSExtra version       | 
| WordPressCS install type | git clone
| IDE (if relevant)        | _N/A_

## Additional Context (optional)

When handling first class callables, the sniff only considers the function name ignoring its namespace if any is present:

https://github.com/WordPress/WordPress-Coding-Standards/blob/20b2c5a58609312f4b815a8d0d9610ecfa308467/WordPress/Sniffs/WP/CronIntervalSniff.php#L157

Different namespace scenarios have varying complexity to fix.. In some cases, it will be necessary to track the current namespace context making the solution more complex. In some cases, this won't be necessary.

## Tested Against `develop` Branch?
- [x] I have verified the issue still exists in the `develop` branch of WordPressCS.

**Comment by rodrigoprimo** [comment]
Just documenting here that I ran a few more tests on this issue in the context of preparing WPCS for PHPCS 4.x and noticed that it only occurs with PHPCS 3.x. Once support for PHPCS 3.x is dropped, we can probably close this issue. The only case that would still be a bit problematic is the following test:

https://github.com/WordPress/WordPress-Coding-Standards/blob/013dc8104e543d39af5d693cc9c1be39d8ea6307/WordPress/Tests/WP/CronIntervalUnitTest.inc#L319

When running this sniff with PHPCS 4.x, it incorrectly flags this code as `WordPress.WP.CronInterval.ChangeDetected`, even though there should be no warning. Fixing it is a bit more involved, as it will require the sniff to resolve relative namespaces, which can probably be addressed in a separate issue.
